src/components/ToggleSwitch.js

```diff
@@ -47,23 +56,23 @@ const CheckBox = styled.input`
   }
 `;
 
-const ToggleSwitch = ({ settings, toggleDarkmode }) => {
-  const toggle = settings.darkmode;
+const ToggleSwitch = ({ id, active, onChangeSettings }) => {
+  const [isChecked, setIsChecked] = React.useState(active);
 
-  function handleChange(event) {
-    toggleDarkmode(event.target.checked);
+  function handleChange() {
+    setIsChecked(!isChecked);
+    onChangeSettings(id, isChecked);
   }
 
   return (
     <CheckBoxWrapper>
       <CheckBox
-        id="checkbox"
+        id={id}
         type="checkbox"
-        toggle={toggle}
+        checked={isChecked}
         onChange={handleChange}
-        checked={toggle}
       />
-      <CheckBoxLabel htmlFor="checkbox" />
+      <CheckBoxLabel htmlFor={id}></CheckBoxLabel>
```

Comment:
Instead of using `id` and `htmlFor`, you could add the `input` as children of the `label`. I was a little bit confused by the `id` property of the `ToggleSwitch` component.

Reply:
I sourced the `ToggleSwitch` from https://codesandbox.io/s/6v7n1vr8yn and I can't make `input` work correctly as a child of the `label`. If I do it like you said functionality still works, obviously, however it won't change color and position. I am assuming this has to do with `&:checked + ${CheckBoxLabel}` in line 45, but I don't fully understand it yet. Will do research and fix it in another PR